runtime/jcl/common/thread.cpp

```diff
@@ -589,6 +589,44 @@ Java_java_lang_Thread_setExtentLocalCache(JNIEnv *env, jclass unusedClass, jobje
 #endif /* JAVA_SPEC_VERSION >= 19 */
 
 #if JAVA_SPEC_VERSION >= 20
+static UDATA findScopedValueBindingsWalkFunction(J9VMThread *currentThread, J9StackWalkState *walkState)
+{
+	UDATA rc = J9_STACKWALK_KEEP_ITERATING;
+	if (NULL != walkState->userData1) {
+		rc = J9_STACKWALK_STOP_ITERATING;
+	} else if (NULL != walkState->method) {
+		J9JavaVM *vm = currentThread->javaVM;
+		J9Method *method = walkState->method;
+		J9Class *methodClass = J9_CLASS_FROM_METHOD(method);
+		J9ROMMethod *romMethod = J9_ROM_METHOD_FROM_RAM_METHOD(method);
+		char *methodName = (char *)J9UTF8_DATA(J9ROMMETHOD_NAME(romMethod));
+
+		if (((J9VMJAVALANGTHREAD(vm) == methodClass)
+			|| (J9VMJAVALANGVIRTUALTHREAD(vm) == methodClass)
+			|| (J9VMJDKINCUBATORCONCURRENTSCOPEDVALUECARRIER(vm) == methodClass))
+			&& (0 == strncmp(methodName, "runWith", 7))
+		) {
+			if (NULL != walkState->arg0EA) {
```

Comment:
I believe that's true (excluding the end-of-stack marker frame, which you will not encounter with a visible frames walk).

Reply:
This will not work for any JITted frame - without special handling, the contents of the arg0EA slot in compiled methods is unreliable. Perhaps this should operate under the same restrictions as `doPrivileged`? On the call we had discussed an alternate solution where the JCL code manually stacks these things to avoid the stack walk, but I'm not familiar with the precise APIs involved, so that might be impractical.